File: backend/src/seed.ts
import mongoose from 'mongoose';
import dotenv from 'dotenv';
import { 
  RoadmapRole, Assessment, Internship, 
  Company, Event, Reward, SoftSkillLesson 
} from './models';

dotenv.config();

const seed = async () => {
  try {
    await mongoose.connect(process.env.MONGODB_URI || 'mongodb://localhost:27017/eduroute');
    console.log('Connected to DB for seeding...');

    // Clear existing data
    await RoadmapRole.deleteMany({});
    await Assessment.deleteMany({});
    await Internship.deleteMany({});
    await Company.deleteMany({});
    await Event.deleteMany({});
    await Reward.deleteMany({});
    await SoftSkillLesson.deleteMany({});

    // Seed Companies
    const companies = await Company.insertMany([
      { name: 'TechFlow Systems', description: 'AI-first engineering company', cultureVideoUrls: ['https://example.com/video1'] },
      { name: 'DataScale AI', description: 'Data analytics platform', cultureVideoUrls: ['https://example.com/video2'] }
    ]);

    // Seed Roadmaps
    await RoadmapRole.create({
      name: 'Frontend Developer',
      slug: 'frontend',
      description: 'Master modern web development',
      modules: [
        { 
          title: 'Basics', 
          level: 'Beginner', 
          tasks: [{ title: 'HTML/CSS', description: 'Learn structure and style' }]
        },
        { 
          title: 'Advanced React', 
          level: 'Intermediate', 
          tasks: [{ title: 'Hooks', description: 'Master React hooks' }]
        }
      ]
    });

    // Seed Internships
    await Internship.insertMany([
      { role: 'Frontend Intern', companyId: companies[0]._id, location: 'Remote', stipend: '₹25,000', duration: '6 months', tags: ['React'], isVerified: true },
      { role: 'Data Science Intern', companyId: companies[1]._id, location: 'Pune', stipend: '₹30,000', duration: '3 months', tags: ['Python'], isVerified: true }
    ]);

    // Seed Assessments
    await Assessment.create({
      title: 'React Fundamentals',
      category: 'Frontend',
      questions: [
        { text: 'What is JSX?', options: ['A JS extension', 'A CSS framework', 'A database', 'None'], correctOption: 0 }
      ],
      points: 200
    });

    // Seed Events
    await Event.create({
      title: 'Global AI Summit',
      city: 'Delhi',
      date: new Date('2024-12-15'),
      monthTag: 'Dec',
      image: 'https://images.unsplash.com/photo-1540575861501-7ad05823c9f5'
    });

    // Seed Rewards
    await Reward.create({
      title: '50% Off IIT Jodhpur Courses',
      description: 'Unlock exclusive academic benefits',
      pointsRequired: 5000,
      partner: 'IIT Jodhpur',
      category: 'Education'
    });

    console.log('Seeding complete!');
    process.exit(0);
  } catch (err) {
    console.error('Seeding error:', err);
    process.exit(1);
  }
};

seed();

File: backend/seed.go
package main

import (
	"encoding/json"
	"fmt"
)

func (s *Server) seedData() error {
	if _, err := s.db.Exec("SET FOREIGN_KEY_CHECKS = 0"); err != nil {
		return err
	}
	defer s.db.Exec("SET FOREIGN_KEY_CHECKS = 1")
	for _, table := range []string{"attempts", "internships", "companies", "roadmaps", "assessments", "events", "rewards", "soft_skill_lessons"} {
		if _, err := s.db.Exec("TRUNCATE TABLE " + table); err != nil {
			return fmt.Errorf("truncate %s: %w", table, err)
		}
	}
	videoOne, _ := json.Marshal([]string{"https://example.com/video1"})
	videoTwo, _ := json.Marshal([]string{"https://example.com/video2"})
	companies, err := s.db.Exec("INSERT INTO companies (name,description,culture_video_urls) VALUES (?,?,?),(?,?,?)", "TechFlow Systems", "AI-first engineering company", videoOne, "DataScale AI", "Data analytics platform", videoTwo)
	if err != nil {
		return err
	}
	firstCompany, err := companies.LastInsertId()
	if err != nil {
		return err
	}
	modules, _ := json.Marshal([]map[string]any{{"title": "Basics", "level": "Beginner", "tasks": []map[string]string{{"title": "HTML/CSS", "description": "Learn structure and style"}}}, {"title": "Advanced React", "level": "Intermediate", "tasks": []map[string]string{{"title": "Hooks", "description": "Master React hooks"}}}})
	if _, err = s.db.Exec("INSERT INTO roadmaps (name,slug,description,modules_json) VALUES (?,?,?,?)", "Frontend Developer", "frontend", "Master modern web development", modules); err != nil {
		return err
	}
	tagsOne, _ := json.Marshal([]string{"React"})
	tagsTwo, _ := json.Marshal([]string{"Python"})
	if _, err = s.db.Exec("INSERT INTO internships (role,company_id,location,stipend,duration,tags,is_verified) VALUES (?,?,?,?,?,?,TRUE),(?,?,?,?,?,?,TRUE)", "Frontend Intern", firstCompany, "Remote", "₹25,000", "6 months", tagsOne, "Data Science Intern", firstCompany+1, "Pune", "₹30,000", "3 months", tagsTwo); err != nil {
		return err
	}
	questions, _ := json.Marshal([]map[string]any{{"text": "What is JSX?", "options": []string{"A JS extension", "A CSS framework", "A database", "None"}, "correctOption": 0}})
	if _, err = s.db.Exec("INSERT INTO assessments (title,category,questions_json,points) VALUES (?,?,?,?)", "React Fundamentals", "Frontend", questions, 200); err != nil {
		return err
	}
	if _, err = s.db.Exec("INSERT INTO events (title,city,event_date,month_tag,image) VALUES (?,?,?,?,?)", "Global AI Summit", "Delhi", "2024-12-15", "Dec", "https://images.unsplash.com/photo-1540575861501-7ad05823c9f5"); err != nil {
		return err
	}
	if _, err = s.db.Exec("INSERT INTO rewards (title,description,points_required,partner,category) VALUES (?,?,?,?,?)", "50% Off IIT Jodhpur Courses", "Unlock exclusive academic benefits", 5000, "IIT Jodhpur", "Education"); err != nil {
		return err
	}
	return nil
}
